cli/azd/internal/repository/infra_confirm.go

```diff
@@ -226,7 +226,7 @@ func getDatabaseNameFromProjectMetadata(detect *detectConfirm, database appdetec
 	result := ""
 	for _, service := range detect.Services {
 		name := service.Metadata.DatabaseNameInPropertySpringDatasourceUrl[database]
-		if name != result {
+		if name != "" {
```

Comment:
This is a bug before current PR: In multi-module project, one module configured database name, and another module, the project name is not configured, the detected database name will not be used before current PR.